### ISSUE TITLE
Non local for variable

### ISSUE THREAD

**Issue by jpic** [koalaman/shellcheck#568]
Should shellcheck prevent leaking of for loop context variables in functions ? Ie. enforce

```bash
    foo() {
        local i  # fail if this line is missing
        for i in 0 1 2
```

Thanks for sharing shellcheck, keep up the great work !!

**Comment by bittorf** [comment]
this sounds like #505

**Comment by LukeShu** [comment]
To put this in to the normal format:

### Feature request: Complain about non-local for-loop variables

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related


#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
foo() {
	#local i # needs to be uncommented
	for i in 0 1 2; do
		echo "$i"
	done
}
foo
```
#### Here's what shellcheck currently says:

```
No issues detected!
```

#### Here's what I wanted or expected to see:

```
Line 4:
        for i in 0 1 2; do
            ^-- SC????: Declare with 'local' or 'declare' to prevent non-local assignment.
```

I have seen other developers (including a younger version of myself) stupidly assume that these variables are implicitly 'local', and omit the 'local' declaration.
